src/mongoserver.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2021 Canonical Ltd
+# See LICENSE file for licensing details.
+
+import secrets
+import string
+
+from tenacity import retry, stop_after_attempt, wait_exponential, RetryError
+from pymongo import MongoClient
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+# We expect the mongodb container to use the
+# default ports
+MONGODB_PORT = 27017
+
+
+class MongoDB:
+    def __init__(self, config):
+        self.app_name = config["app_name"]
+        self.replica_set_name = config["replica_set_name"]
+        self.num_peers = config["num_peers"]
+        self.port = config["port"]
+        self.root_password = config["root_password"]
+        self.unit_ips = config["unit_ips"]
+
+    def client(self) -> MongoClient:
+        """Construct a client for the MongoDB database.
+
+        The timeout for all queries using this client object is 1 sec.
+
+        Returns:
+            A pymongo :class:`MongoClient` object.
```

Comment:
```suggestion
            A pymongo `MongoClient` object.
```

This notation isn't broadly used elsewhere in your code so I'd not bother for now :)